xdc-plugin/xdc.cc

```diff
@@ -98,15 +98,15 @@ struct GetPorts : public Pass {
 		port_name = args.at(1);
 		char port[128];
 		int bit(0);
-		std::string port_signal(port_name);
-		if (sscanf(port_name.c_str(), "%[^[][%d]", port, &bit) == 2) {
-			port_signal = std::string(port);
+		if (!sscanf(port_name.c_str(), "%[^[][%d]", port, &bit)) {
+			log_error("Couldn't find port %s\n", port_name.c_str());
 		}
+		std::string port_signal(port);
 
 		RTLIL::IdString port_id(RTLIL::escape_id(port_signal));
 		if (auto wire = top_module->wire(port_id)) {
-			if (bit >= wire->start_offset && bit < wire->start_offset + wire->width) {
-				if (isInputPort(wire) || isOutputPort(wire)) {
+			if (isInputPort(wire) || isOutputPort(wire)) {
```

Comment:
It should not hit us but what about "isInoutPort(wire)"?

Reply:
Inout wire will return true for both checks so I guess we are good here, but I will double check and update the test accordingly.

Reply:
Inout are handled correctly. Added one to the test case.